packages/shared/lib/services/connections/credentials/refresh.ts

```diff
@@ -78,6 +78,14 @@ export async function refreshOrTestCredentials(props: RefreshProps): Promise<Res
 
         span.setTag('connectionId', props.connection.connection_id).setTag('authType', props.connection.credentials.type);
 
+        // TODO: remove this when cron is using other columns
+        await connectionService.updateLastFetched(props.connection.id);
+
+        // short-circuit if we know the refresh will fail
+        if (props.connection.refresh_exhausted && !props.instantRefresh) {
+            return Ok(props.connection);
+        }
+
```

Comment:
would it make more sense to have this logic in `shouldRefreshCredentials`?

Reply:
it's a global flag, shouldRefresh only applies to refreshable connections whereas this block refresh and test.